pulsar-broker/src/test/java/org/apache/pulsar/client/impl/MessageChecksumTest.java

```diff
@@ -69,7 +69,7 @@ protected void cleanup() throws Exception {
     @Override
     protected void customizeNewPulsarClientBuilder(ClientBuilder clientBuilder) {
         // disable connection pooling
-        clientBuilder.connectionsPerBroker(0);
+        // clientBuilder.connectionsPerBroker(0);
```

Comment:
please revert

Reply:
The value of `connectionsPerBroker` must be greater than 0 (doc in [ClientBuilder L266](https://github.com/apache/pulsar/blob/master/pulsar-client-api/src/main/java/org/apache/pulsar/client/api/ClientBuilder.java#L266)), and this test will fail after the [check](https://github.com/apache/pulsar/pull/11973/files#diff-325e5f133e1f77dace278a691a571b3912dc6c44c0a6805f1c9b93f71f12dc7dR172) is added, so a comment is made.

Reply:
so what do you suggest ?
modifying a test means that we modify the behaviour and this is not good (most of the times)

This test tells me that 0 is an allowed number, so you have to update your preconditions (and docs?) and let this test pass

Reply:
I agree with you that 0 is a feasible value for this test. So I update the document and revert the test.

Reply:
I took a look at how this setting is used, and 0 is a valid setting. https://github.com/apache/pulsar/blob/4147db88bc08eb3b2dab97a45af178fd9f4c7a6b/pulsar-client/src/main/java/org/apache/pulsar/client/impl/ConnectionPool.java#L148-L151